presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/DriverConnectionFactory.java

```diff
@@ -45,20 +54,41 @@ public static Properties basicConnectionProperties(BaseJdbcConfig config)
         return connectionProperties;
     }
 
-    public DriverConnectionFactory(Driver driver, String connectionUrl, Properties connectionProperties)
+    public DriverConnectionFactory(Driver driver, String connectionUrl, Optional<String> userCredentialKey, Optional<String> passwordCredentialKey, Properties connectionProperties)
     {
         this.driver = requireNonNull(driver, "driver is null");
         this.connectionUrl = requireNonNull(connectionUrl, "connectionUrl is null");
         this.connectionProperties = new Properties();
         this.connectionProperties.putAll(requireNonNull(connectionProperties, "basicConnectionProperties is null"));
+        this.userCredentialKey = requireNonNull(userCredentialKey, "userCredentialKey is null");
+        this.passwordCredentialKey = requireNonNull(passwordCredentialKey, "passwordCredentialKey is null");
     }
 
     @Override
     public Connection openConnection(JdbcIdentity identity)
             throws SQLException
     {
+        userCredentialKey.ifPresent(userKey -> setConnectionProperty(connectionProperties, identity.getExtraCredentials(), userKey, "user"));
+        passwordCredentialKey.ifPresent(passwordKey -> setConnectionProperty(connectionProperties, identity.getExtraCredentials(), passwordKey, "password"));
+
+        userCredentialKey.ifPresent(userKey -> {
```

Comment:
This code looks duplicated. I'm guessing you extracted a method for the above calls and forgot remove this and the password one?